tests/test_models.py

```diff
@@ -58,3 +63,39 @@ def test_calculate_support_status_system(mocker, current_date, system_start, sys
 )
 def test_get_rhel_display_name(name, major, minor, expected):
     assert _get_rhel_display_name(name, major, minor) == expected
+
+
+@pytest.mark.parametrize("count", (0, 1, 2))
+def test_system_populate_systems_from_systems_details(generate_system_detail, count):
+    system_ids = set()
+    systems_detail = set()
+
+    for _ in range(0, count):
+        system, system_id = generate_system_detail
+        system_ids.add(system_id)
+        systems_detail.add(system)
+
+    system_lifecycle = System(
+        name="RHEL",
+        major=9,
+        minor=1,
+        lifecycle_type=LifecycleType.mainline,
+        count=count,
+        start_date=date(2022, 5, 17),
+        end_date=date(2032, 5, 31),
+        systems_detail=systems_detail,
+    )
+
+    assert system_lifecycle.systems == system_ids
+
+
+def test_get_system_uuids():
```

Comment:
What is the purpose of this test?

Reply:
Will improve the name. It's testing case when any of the attributes is available (`data.get("systems_detail", []) or data.get("potentiallyAffectedSystemsDetail", [])`) - mostly to get good coverage.

Reply:
@samdoran changed